lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -634,11 +634,7 @@ public function getColumns()
         }
         $colNames = array_unique(array_merge($pkCols, $fkCols, array_keys($columns)));
 
-        uksort($columns, function ($a, $b) use ($colNames) {
-            return (array_search($a, $colNames) >= array_search($b, $colNames));
-        });
-
-        return $columns;
+        return array_replace(array_flip($colNames), $columns);
```

Comment:
@verheyenkoen both mehods are called 1 time per getColumns() invocation (as many times as in proposal you made)